src/login.service.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  BadRequestException,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { User } from './entities/user.entity';
+
+@Injectable()
+export class LoginService {
+  constructor(
+    @InjectRepository(User)
+    private usersRepository: Repository<User>,
+    private jwtService: JwtService,
+  ) {}
+
+  async login(username: string, password: string) {
```

Comment:
I suggest adding a return type as `Jwt` interface.

Reply:
JWTResult added as a type aiming to keep the project styleguides. Can you have a look on 60ef95daafe1c101fff4da7990810d1c4c072f21 and tell me what you think?

Reply:
Looks great but already requested `JwtResult` since everyone is using in this way.